### PR TITLE
Remove doc scroll listener after navigating away

### PR DESCRIPTION
**Pull request by hunicus**
Listener was activated but never removed, so it remained active even after navigating to other pages where it wasn't needed.

### CODE REVIEW

**Review by softsimon — APPROVED** [review]
I can't see what problem this caused but memory leak fixes are always welcome

tested ACK

v2.5.0-dev [[e0c61f72](https://github.com/mempool/mempool/commit/e0c61f72)]

**Comment by hunicus** [comment]
Just for performance. On its own, not terrible, but don't want them to pile up.

File: frontend/src/app/docs/api-docs/api-docs.component.ts [hunk 1]
```diff
@@ -87,6 +85,14 @@ export class ApiDocsComponent implements OnInit, AfterViewInit {
     });
   }
 
+  ngOnDestroy(): void {
```
Comment (Arooba-git) [inline frontend/src/app/docs/api-docs/api-docs.component.ts:88]: Hm, we didn't import or implement `OnDestroy` from `angular/core` 🤔 , can we use `ngOnDestroy` without doing that? 🤔

Comment (softsimon) [reply frontend/src/app/docs/api-docs/api-docs.component.ts:88]: We can't, but this changed after merge and is not present in master.

Comment (Arooba-git) [reply frontend/src/app/docs/api-docs/api-docs.component.ts:88]: @softsimon Ah, thanks for the clarification.. didn't use angular for years so just wanted to confirm 👍  :)

**Pull request merged**